__tests__/__integration__/imperative/__tests__/__integration__/cli/config/init/cli.imperative-test-cli.config.init.integration.test.ts

```diff
@@ -0,0 +1,381 @@
+    });
+    it("should initialize a global config with prompting", () => {
+        const response = runCliScript(__dirname + "/__scripts__/init_config_prompt.sh",
+            TEST_ENVIRONMENT.workingDir, ["--global"]);
+        const expectedConfigLocation = path.join(TEST_ENVIRONMENT.workingDir, "imperative-test-cli.config.json");
+        const expectedSchemaLocation = path.join(TEST_ENVIRONMENT.workingDir, "schema.json");
+        const expectedConfigObject: IConfig = {
+            $schema: "./schema.json",
+            profiles: {
+                my_secured: {
+                    type: "secured",
+                    properties: {
+                        info: ""
+                    }
+                }
+            },
+            defaults: {
+                secured: "my_secured"
+            },
+            plugins: [],
+            secure: ["profiles.my_secured.properties.secret"]
+        };
+        expect(response.output.toString()).toContain(`Saved config template to`);
+        expect(response.output.toString()).toContain(expectedConfigLocation);
+        expect(fs.existsSync(expectedConfigLocation)).toEqual(true);
+        expect(fs.existsSync(expectedSchemaLocation)).toEqual(true);
+        expect(JSON.parse(fs.readFileSync(expectedConfigLocation).toString())).toEqual(expectedConfigObject);
+        expect(JSON.parse(fs.readFileSync(expectedSchemaLocation).toString())).toEqual(expectedSchemaObject);
+    });
+    it("should initialize a user global config with prompting", () => {
+        const response = runCliScript(__dirname + "/__scripts__/init_config_prompt.sh",
+            TEST_ENVIRONMENT.workingDir, ["--global --user"]);
+        const expectedConfigLocation = path.join(TEST_ENVIRONMENT.workingDir, "imperative-test-cli.config.user.json");
+        const expectedSchemaLocation = path.join(TEST_ENVIRONMENT.workingDir, "schema.json");
+        const expectedConfigObject: IConfig = {
+            $schema: "./schema.json",
+            profiles: {
+                my_secured: {
+                    type: "secured",
+                    properties: {
+                        info: ""
+                    }
+                }
+            },
+            defaults: {
+                secured: "my_secured"
+            },
+            plugins: [],
+            secure: ["profiles.my_secured.properties.secret"]
+        };
+        expect(response.output.toString()).toContain(`Saved config template to`);
+        expect(response.output.toString()).toContain(expectedConfigLocation);
+        expect(fs.existsSync(expectedConfigLocation)).toEqual(true);
+        expect(fs.existsSync(expectedSchemaLocation)).toEqual(true);
+        expect(JSON.parse(fs.readFileSync(expectedConfigLocation).toString())).toEqual(expectedConfigObject);
+        expect(JSON.parse(fs.readFileSync(expectedSchemaLocation).toString())).toEqual(expectedSchemaObject);
+    });
+    // it("should create a profile of a specified name", () => {
```

Comment:
Commented test?

Reply:
It's commented because the CLI handler has not been updated when that option is invoked yet - it is one of the later use cases though, so we wanted to keep it around for now.